packages/nimble-components/src/rich-text-editor/specs/README.md

```diff
@@ -0,0 +1,401 @@
+    view the content of the editor.
+-   The minimum and maximum width of the `editor section` will be fixed value considering the mobile and desktop view of the component respectively. However, if the
+    screen width is reduced
+    below the specified minimum width, the layout may not handle or adapt to the content.
+-   The `footer section` will be a flexbox container and have the same width as the editor section.
+-   The `formatting option toolbar` in the footer section will be enclosed within a flexbox container, enabling the buttons to wrap and occupy roughly seventy
+    percent of the entire footer. If the buttons exceed the seventy percent limit, they will be positioned below in the same container, aligning from left
+    to right. The remaining thirty percent of the footer will be enclosed in another flexbox container dedicated to `footer-actions` slot elements.
+
+### Anatomy
+
+_Shadow DOM template_
+
+```html
+<template>
+    <section id="editor"></section>
+    <footer>
+        <section id="toolbar">
+            <nimble-toolbar>
+                <nimble-button></nimble-button>
+            </nimble-toolbar>
+            <slot name="footer-actions"></slot>
+        </section>
+    </footer>
+</template>
+```
+
+_Slot Names_
+
+-   `footer-actions`:
+    1. It is a container that allows a client to easily place buttons at the right bottom of the component to interact with the editor.
+    2. If no content is slotted in the `footer-actions`, the element will be emptied and shrunk to accommodate buttons from the
+       toolbar.
+    3. If the content is slotted in the footer-actions, it will occupy a maximum of thirty percent of the entire horizontal footer. If
+       there are additional elements beyond this limit, they will be positioned below within the same footer-actions container.
+
+_Note_: The positioning of these slot elements in the mobile view of the component has not yet been confirmed.
+
+_Host Classes_
+
+-   none
+
+_Slotted Content/Slotted Classes_
+
+-   none
+
+_CSS Parts_
+
+-   none
+
+### `nimble-rich-text-viewer`
+
+The `nimble-rich-text-viewer` is used for viewing rich text content when a markdown string is passed to it. It performs the post-processing
+tasks to convert the markdown string to corresponding HTML nodes for each text formatting.
+
+### API
+
+_Props/Attrs_
+
+-   `markdownValue` - is an accessor used to get and set the markdown value.
+    -   `getter` - this will merely return the markdown string that is set to the component.
+    -   `setter` - this will parse the markdown string into a Node using
+        [prosemirror-markdown parser](https://github.com/ProseMirror/prosemirror-markdown/blob/master/src/from_markdown.ts#L199) and convert to an HTML to
+        render it in the component section.
+
+_Methods_
+
+-   none
+
+_Events_
+
+-   none
+
+_CSS Classes and CSS Custom Properties that affect the component_
+
+-   The height of the component will always grow to fit the content.
```

Comment:
It would be good to clarify: clients can still set the width? And this setting could affect the height (a smaller width will cause the content to reflow and take up more height)?

Reply:
Yes, it will increase the height if the width is small. I have updated the same in the spec.